pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -326,15 +326,19 @@ func (m *managerImpl) processShutdownEvent() error {
 			}(pod, group)
 		}
 
-		c := make(chan struct{})
+		var (
+			doneCh = make(chan struct{})
+			timer  = time.NewTimer(time.Duration(group.ShutdownGracePeriodSeconds) * time.Second)
+		)
 		go func() {
-			defer close(c)
+			defer close(doneCh)
 			wg.Wait()
 		}()
 
 		select {
-		case <-c:
-		case <-time.After(time.Duration(group.ShutdownGracePeriodSeconds) * time.Second):
+		case <-doneCh:
+			timer.Stop()
+		case <-timer.C:
```

Comment:
timer.C is a little confusing, would it be possible to change this to timer.After, from here in the docs? https://pkg.go.dev/time

Reply:
A time.After can't be cancelled so it has to run the full time specified.